### PR TITLE
Fixbug/289 in user profile picture response 500 internal server error always resolve

### PR DESCRIPTION
**Pull request by NazarVavrushchak**

### CODE REVIEW

File: core/src/main/java/greencity/config/SecurityConfig.java [hunk 1]
```diff
@@ -130,10 +132,12 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 "/ownSecurity/signIn",
                                 "/ownSecurity/updatePassword")
                         .permitAll()
+                        .requestMatchers(HttpMethod.GET, "/user/isOnline/{userId}/")
+                        .authenticated()
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/config/SecurityConfig.java]: .requestMatchers(HttpMethod.GET, "/user/isOnline/{userId}/")
                        .authenticated()
why did you need it here to solve the current task?

File: core/src/main/java/greencity/controller/UserController.java [hunk 2]
```diff
@@ -59,20 +66,26 @@ public class UserController {
...
+        if (principal == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+        try {
+            UserStatusDto updatedStatus = userService.updateStatus(
+                    userStatusDto.getId(), userStatusDto.getUserStatus(), principal.getName());
+            return ResponseEntity.status(HttpStatus.OK).body(updatedStatus);
+        } catch (BadUpdateRequestException | LowRoleLevelException e) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did you modify this method? Task 289 about the bug in updateUserProfilePicture method?

File: core/src/main/java/greencity/controller/UserController.java [hunk 3]
```diff
@@ -368,16 +382,19 @@ public ResponseEntity<UserProfileDtoResponse> getUserProfileInformation(
...
-        return ResponseEntity
-            .status(HttpStatus.OK)
-            .body(userService.checkIfTheUserIsOnline(userId));
+            @Parameter(description = "Id of the user. Cannot be empty.") @PathVariable Long userId) {
+        try {
+            boolean isOnline = userService.checkIfTheUserIsOnline(userId);
+            return ResponseEntity.status(HttpStatus.OK).body(isOnline);
+        } catch (WrongIdException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(false);
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did you modify this method? The bug is in updateUserProfilePicture mrthod.

File: core/src/main/java/greencity/controller/UserController.java [hunk 4]
```diff
@@ -388,15 +405,20 @@ public ResponseEntity<Boolean> checkIfTheUserIsOnline(
...
+            @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
     })
     @GetMapping("/{userId}/profileStatistics/")
     public ResponseEntity<UserProfileStatisticsDto> getUserProfileStatistics(
-        @Parameter(description = "Id of current user. Cannot be empty.") @PathVariable @CurrentUserId Long userId) {
+            @PathVariable Long userId
+            , @Parameter(description = "Id of current user. Cannot be empty.") @CurrentUserId Long currentUserId) {
+        if (!currentUserId.equals(userId)) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did you modify this method? How does it helps to solve the issue from task 289?

File: kalina.jpg
Comment (VitaliiChernenko) [inline kalina.jpg:1]: Why do we need this picture in project?

**Comment by VitaliiChernenko** [comment]
To solve the current issue quite enough is to remove annotation @AuthenticationPrincipal from parameters in updateUserProfilePicture method. It is not necessary to modify all those classes.

File: core/src/main/java/greencity/controller/UserController.java (hunk 4 above)
Comment (NazarVavrushchak) [reply core/src/main/java/greencity/controller/UserController.java]: Sorry)The whole solution in last commit(50d1d59799adadbfbd3e44cea27441d3286f5fe7)

File: kalina.jpg
Comment (NazarVavrushchak) [reply kalina.jpg:1]: To be able to test the method with the provided image and don't download it form the Internet.

**Comment by NazarVavrushchak** [comment]
> To solve the current issue quite enough is to remove annotation @AuthenticationPrincipal from parameters in updateUserProfilePicture method. It is not necessary to modify all those classes.

This is not enough to solve the problem

File: kalina.jpg
Comment (VitaliiChernenko) [reply kalina.jpg:1]: Maybe it would be better to move it somewhere closer to the test classes?
I mean test/resources. And if You write tests for some of services, you can notice that in service/..../test/resources present picture for tests. You can also add your own here or use one that already exists.

Comment (NazarVavrushchak) [reply kalina.jpg:1]: You are right to use the existing image is better

**Review by DmytroLysenko1 — APPROVED** [review]
